### PR TITLE
Add Purchase Order store struct accessors and builders

### PR DESCRIPTION
**Pull request by dnewh**
This adds public accessor methods to the PurchaseOrderStore structs
and makes the structs' fields private. This also adds builders for the
structs. This brings the module in-line with the other Grid stores.

### CODE REVIEW

**Review by shannynalayna — DISMISSED** [review]

**Review by dplumb94 — DISMISSED** [review]

File: sdk/src/purchase_order/store/error.rs [hunk 1]
```diff
@@ -85,3 +85,34 @@ impl From<diesel::r2d2::PoolError> for PurchaseOrderStoreError {
...
+            PurchaseOrderBuilderError::BuildError(err) => Some(&**err),
+        }
+    }
+}
+
+impl fmt::Display for PurchaseOrderBuilderError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            PurchaseOrderBuilderError::MissingRequiredField(ref s) => {
+                write!(f, "failed to build purchase order: {}", s)
```
Comment (shannynalayna) [inline sdk/src/purchase_order/store/error.rs:111]: I feel like this error msg should be a little more specific to the variant, something like `Missing required field: {}` and then setting up the error with the variant name.

File: sdk/src/purchase_order/store/error.rs [hunk 2]
```diff
@@ -85,3 +85,34 @@ impl From<diesel::r2d2::PoolError> for PurchaseOrderStoreError {
...
+            }
+            PurchaseOrderBuilderError::BuildError(ref s) => {
+                write!(f, "failed to build purchase order: {}", s)
```
Comment (shannynalayna) [inline sdk/src/purchase_order/store/error.rs:114]: Should this error specifically refer to `purchase order`, won't other structs have to raise this error if they encounter it? Maybe remove the `purchase order` part of this msg

**Review by shannynalayna — APPROVED** [review]

**Review by shannynalayna — APPROVED** [review]

File: sdk/src/purchase_order/store/mod.rs [hunk 3]
```diff
@@ -38,29 +38,385 @@ impl PurchaseOrderList {
...
+    accepted_version_id: String,
+    versions: Vec<PurchaseOrderVersion>,
+    created_at: i64,
+    start_commit_num: i64,
+    end_commit_num: i64,
+    service_id: Option<String>,
+}
+
+impl PurchaseOrderBuilder {
+    /// Sets the UUID for this PO
```
Comment (vaporos) [inline sdk/src/purchase_order/store/mod.rs:127]: s/UUID/unique id/

Comment (dnewh) [reply sdk/src/purchase_order/store/mod.rs]: updated

**Review by shannynalayna — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**